Tools/gsc/test/makefile

```diff
@@ -0,0 +1,138 @@
+TESTCASES?=python3 hello-world nodejs bash numpy pytorch
+DISTRIBUTIONS?=ubuntu18.04 ubuntu16.04
+TESTS=$(foreach D,$(DISTRIBUTIONS),$(foreach T,$(TESTCASES),$D-$T))
+SGX_DEVICES=--device=/dev/gsgx --device=/dev/isgx -v /var/run/aesmd/aesm.socket:/var/run/aesmd/aesm.socket
+
+.PHONY: test clean cleanBase cleanGSC cleanContainer
+
+IMAGE_SUFFIX?=
+
+all: $(TESTS)
+	for d in $(DISTRIBUTIONS); do \
+		sed 's/\"distro\": \".*\"/\"distro\": \"'$${d}'\"/' ../config.json.template > ../config.json; \
+		$(MAKE) $(addprefix gsc-$${d}-, $(TESTCASES)) || exit 1; \
+	done
+
+%: %.dockerfile
+	echo "Build base image $@.dockerfile"
+	docker build -t $(addsuffix $(IMAGE_SUFFIX), $@) -f $@.dockerfile .
+
+gsc-%-bash:
+	echo "Build graphenized image $@"
+	cd .. && ./gsc build $(addsuffix $(IMAGE_SUFFIX), $*-bash) bash.manifest ls.manifest
+
+gsc-%-python3: %-python3.manifest
+	echo "Build graphenized image $@"
+	cd .. && ./gsc build $(addsuffix $(IMAGE_SUFFIX), $*-python3) $(addprefix test/, $^ sh.manifest ls.manifest)
+
+gsc-%-pytorch: %-pytorch.manifest
+	echo "Build graphenized image $@"
+	cd .. && ./gsc build $(addsuffix $(IMAGE_SUFFIX), $*-pytorch) $(addprefix test/, $^ sh.manifest uname.manifest)
+
+gsc-%:
+	echo "Build graphenized image $@"
+	cd .. && ./gsc build $(addsuffix $(IMAGE_SUFFIX), $*) test/$(*:gsc-%=%).manifest
+
+test: $(addprefix test-, $(DISTRIBUTIONS))
+	echo "[SUCCESS] Completed all GSC test cases"
+
+MAXTESTNUM?=11
+
+test-1-%:
+	docker run $(SGX_DEVICES) $(addsuffix $(IMAGE_SUFFIX), gsc-$*-python3) -c 'print("HelloWorld!")' >out 2>/dev/null;
```

Comment:
It will be better to have a separate image for each testcase. In this case, the gsc-image -> gsc-ubuntu18.04-python3, still builds and signs sh program and ls program, which is not needed for printing HelloWorld.